Front End/src/app/attendance/attendance.component.spec.ts

```diff
@@ -0,0 +1,25 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { AttendanceComponent } from './attendance.component';
+
+describe('AttendantsComponent', () => {
+  let component: AttendanceComponent;
+  let fixture: ComponentFixture<AttendanceComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [ AttendanceComponent ]
```

Comment:
You don't have a provider here for any of the services injected into your component, so these tests will fail